clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1139,6 +1164,13 @@ public CreateTopicsResult createTopics(final Collection<NewTopic> newTopics,
             @Override
             public void handleResponse(AbstractResponse abstractResponse) {
                 CreateTopicsResponse response = (CreateTopicsResponse) abstractResponse;
+                // Check for controller change
+                for (ApiError error : response.errors().values()) {
+                    if (error.error() == Errors.NOT_CONTROLLER) {
+                        runnable.shouldRequestMetadata.set(true);
+                        throw error.exception();
```

Comment:
I guess I would have expected that these transient errors would get retried internally. Is the intent to do this separately?

Reply:
They can't be retried internally because we have no generic top level error code.  So if createTopics fails, you get back a response that has {foo : NOT_CONTROLLER, bar: NOT_CONTROLLER, baz: NOT_CONTROLLER, etc.}  You can't parse this without knowing the subtype, which means it has to be done here.

Reply:
To clarify my question, does the user request fail because of a transient NOT_CONTROLLER error or do we retry the request? I had expected we would retry, but perhaps we are leaving that for future work?